contracts/vault/src/execute.rs

```diff
@@ -0,0 +1,464 @@
+        .filter_map(|market_id_res| {
+            let market_id = market_id_res.ok()?;
+
+            let resp: StatusResp = deps
+                .querier
+                .query(&QueryRequest::Wasm(WasmQuery::Smart {
+                    contract_addr: market_id.clone(),
+                    msg: to_json_binary(&MarketQueryMsg::Status { price: None }).ok()?,
+                }))
+                .ok()?;
+            if let Token::Native { denom, .. } = &resp.collateral {
+                if denom == &config.usdc_denom {
+                    Some(Ok(market_id))
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect::<Result<Vec<String>>>()?;
+
+    let messages: Vec<CosmosMsg> = markets
+        .iter()
+        .filter_map(|market| {
+            Some(
+                WasmMsg::Execute {
+                    contract_addr: market.to_string(),
+                    msg: to_json_binary(&MarketExecuteMsg::ClaimYield {}).ok()?,
+                    funds: vec![],
+                }
+                .into(),
+            )
+        })
+        .collect();
+
+    Ok(Response::new()
+        .add_messages(messages)
+        .add_attribute("action", "collect_yield")
+        .add_attribute("markets_processed", markets.len().to_string()))
+}
+
+fn execute_process_withdrawal(deps: DepsMut, env: Env, _info: MessageInfo) -> Result<Response> {
+    let config = state::CONFIG.load(deps.storage)?;
+
+    if config.paused {
+        return Err(anyhow!("The contract is paused"));
+    }
+
+    let vault_balance = deps
+        .querier
+        .query_balance(&env.contract.address, &config.usdc_denom)?
+        .amount;
+
+    let mut processed_amount = Uint128::zero();
+    let limit = 20;
+    let mut messages: Vec<CosmosMsg> = Vec::with_capacity(limit);
+    let mut processed_entries: Vec<(QueueId, Addr)> = Vec::with_capacity(limit);
+
+    for item in state::WITHDRAWAL_QUEUE
+        .range(deps.storage, None, None, Order::Ascending)
+        .take(limit)
+    {
+        let (id, request) = item?;
+
+        processed_amount += request.amount;
+        if processed_amount > vault_balance {
+            break;
+        }
+
+        messages.push(
+            BankMsg::Send {
+                to_address: request.user.to_string(),
+                amount: vec![Coin {
+                    denom: config.usdc_denom.clone(),
+                    amount: request.amount,
+                }],
+            }
+            .into(),
+        );
+
+        processed_entries.push((id, request.user.clone()));
+    }
+
+    for (id, user) in &processed_entries {
+        state::WITHDRAWAL_QUEUE.remove(deps.storage, id.clone());
+        state::USER_WITHDRAWALS.remove(deps.storage, (user, id.clone()));
+    }
+
+    if !processed_amount.is_zero() {
+        if let Ok(mut total) = state::TOTAL_PENDING_WITHDRAWALS.load(deps.storage) {
```

Comment:
I'd rather this was not conditional - instead of the `if let Ok(...)` simply use the `?` instead. At this point we know the `load` call cannot fail - if it did that would indicate a bug somewhere.